cmd/network/verification.go

```diff
@@ -0,0 +1,101 @@
+package network
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	ocmlog "github.com/openshift-online/ocm-sdk-go/logging"
+	"github.com/openshift/osd-network-verifier/pkg/cloudclient"
+	"github.com/spf13/cobra"
+)
+
+var (
+	defaultTags            = map[string]string{"osd-network-verifier": "owned", "red-hat-managed": "true", "Name": "osd-network-verifier"}
+	regionEnvVarStr string = "AWS_REGION"
+	regionDefault   string = "us-east-2"
+)
+
+type egressConfig struct {
+	vpcSubnetID  string
+	cloudImageID string
+	instanceType string
+	cloudTags    map[string]string
+	debug        bool
+	region       string
+	timeout      time.Duration
+	kmsKeyID     string
+}
+
+func getDefaultRegion() string {
+	val, present := os.LookupEnv(regionEnvVarStr)
+	if present {
+		return val
+	}
+	return regionDefault
+
+}
+func NewCmdValidateEgress() *cobra.Command {
+	config := egressConfig{}
+
+	validateEgressCmd := &cobra.Command{
+		Use:   "verify-egress",
+		Short: "Verify essential openshift domains are reachable from given subnet ID.",
+		Long:  `Verify essential openshift domains are reachable from given subnet ID.`,
+		Example: `For AWS, ensure your credential environment vars 
+AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY (also AWS_SESSION_TOKEN for STS credentials) 
+are set correctly before execution.
+
+# Verify that essential openshift domains are reachable from a given SUBNET_ID
+osdctl network verify-egress --subnet-id $(SUBNET_ID) --image-id $(IMAGE_ID)`,
+		Run: func(cmd *cobra.Command, args []string) {
```

Comment:
For readability, can this function be spun out instead of being anonymously defined?

Reply:
Would be nice, but this is not a far departure from the other cmds in the network package. So it's better to keep pattern consistency, imo.

Reply:
on a second thought, I agree. Will update shortly.

Reply:
FWIW, any other command like this in osdctl should be spun out I think

Reply:
updated